src/wasm/wasm-emscripten.cpp

```diff
@@ -397,6 +397,32 @@ struct RemoveStackPointer : public PostWalker<RemoveStackPointer> {
   Global* stackPointer;
 };
 
+// lld can sometimes produce a build with an imported mutable __stack_pointer
+// (i.e.  when linking with -fpie).  This method internalizes the
+// __stack_pointer and initializes it from an immutable global instead.
+// For -shared builds we instead call replaceStackPointerGlobal.
+void EmscriptenGlueGenerator::internalizeStackPointerGlobal() {
+  Global* stackPointer = getStackPointerGlobal();
+  if (!stackPointer || !stackPointer->imported() || !stackPointer->mutable_) {
+    return;
+  }
+
+  Name internalName = stackPointer->name;
+  Name externalName = internalName.c_str() + std::string("_import");
```

Comment:
how about adding an assert that `stackPointer->imported()`, I think that might help readability too?

Reply:
There is an early return above when its not.

Reply:
Oh sorry, I missed that.